nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-processors/src/main/java/org/apache/nifi/processors/kafka/GetKafka.java

```diff
@@ -481,4 +496,51 @@ private void releaseFlowFile(FlowFile flowFile, ProcessSession session, Map<Stri
             session.transfer(flowFile, REL_SUCCESS);
         }
     }
+
+    @Override
+    public StateMap getState() throws IOException {
+        if (!isReadyToAccessState()) {
+            return null;
+        }
+        final Map<String, String> partitionOffsets = KafkaUtils.retrievePartitionOffsets(zookeeperConnectionString, topic, groupId);
+
+        return new StandardStateMap(partitionOffsets, System.currentTimeMillis());
+    }
+
+    private boolean isReadyToAccessState() {
+        if(StringUtils.isEmpty(zookeeperConnectionString)
+                || StringUtils.isEmpty(topic)
+                || StringUtils.isEmpty(groupId)) {
+            return false;
+        }
+        return true;
+    }
+
+    @Override
+    public void clear() throws IOException {
+        if (!isReadyToAccessState()) {
+            return;
+        }
+        KafkaUtils.clearPartitionOffsets(zookeeperConnectionString, topic, groupId);
```

Comment:
Shouldn't this block the onTrigger from attempting to reach out to Kafka to get new messages?